tests/test_model_builder.py

```diff
@@ -0,0 +1,294 @@
+        for als in pint_par_obj.aliases:
+            assert mb.alias_to_pint_param(als) == pint_par_obj.name
+            try:
+                als_prefix, id, ids = split_prefixed_name(als)
+            except PrefixError:
+                als_prefix = als
+        assert mb.alias_to_pint_param(als_prefix + "2") == pint_par_obj.prefix + "2"
+        assert mb.alias_to_pint_param(als_prefix + "55") == pint_par_obj.prefix + "55"
+
+
+def test_conflict_alias():
+    """Test if model builder detects the alais conflict.
+    """
+    mb = ModelBuilder()
+    # Test conflict parameter alias name
+    with pytest.raises(ConflictAliasError):
+        _ = mb._add_alias_to_map("F0", "F1", mb.param_alias_map)
+    # Define conflict alais from component class
+    class SimpleModel2(PhaseComponent):
+        """Very simple test model component
+        """
+
+        register = True
+        category = "simple_test"
+
+        def __init__(self):
+            super(SimpleModel2, self).__init__()
+            self.add_param(
+                floatParameter(name="TESTPARAMF0", aliases=["F0"], value=0.0, unit="s")
+            )
+
+    with pytest.raises(ConflictAliasError):
+        mb2 = ModelBuilder()
+    del Component.component_types["SimpleModel2"]
+
+
+def test_overlap_component(simple_model_overlap, simple_model_alias_overlap):
+    """Test if model builder detects the overlap component correctly.
+    """
+    mb = ModelBuilder()
+    # Test overlap
+    overlap = mb._get_component_param_overlap(simple_model_overlap)
+    assert "Spindown" in overlap.keys()
+    assert overlap["Spindown"][0] == set(["F0"])
+    # Only one over lap parameter F0
+    # Since the _get_component_param_overlap returns non-overlap part,
+    # we test if the non-overlap number makes sense.
+    assert overlap["Spindown"][1] == len(simple_model_overlap.params) - 1
+    assert overlap["Spindown"][2] == len(mb.components["Spindown"].params) - 1
+
+    a_overlap = mb._get_component_param_overlap(simple_model_alias_overlap)
+    assert a_overlap["Spindown"][0] == set(["F0"])
+    assert a_overlap["Spindown"][1] == len(simple_model_alias_overlap.params) - 1
+    assert a_overlap["Spindown"][2] == len(mb.components["Spindown"].params) - 1
+    assert a_overlap["AstrometryEcliptic"][0] == set(["ELONG"])
+    assert (
+        a_overlap["AstrometryEcliptic"][1] == len(simple_model_alias_overlap.params) - 1
+    )
+    assert (
+        a_overlap["AstrometryEcliptic"][2]
+        == len(mb.components["AstrometryEcliptic"].params) - 1
+    )
+
+
+def test_subset_component(sub_set_model):
+    """Test if model builder detects the subset component.
+    """
+    mb = ModelBuilder()
+    # Test subset
+    superset = mb._is_subset_component(sub_set_model)
+    assert superset == "Spindown"
+
+
+def test_model_from_par():
+    """Test Get model from test par file.
+    """
+    mb = ModelBuilder()
+    param_inpar, repeat = mb.parse_parfile(io.StringIO(test_par1))
```

Comment:
Can you put `test_par1` inline here, so we can see why 14 is the right number of components?